statsmodels/genmod/generalized_linear_model.py

```diff
@@ -714,17 +718,26 @@ def fit(self, start_params=None, maxiter=100, method='IRLS',
                                       tol=tol, scale=scale,
                                       full_output=full_output,
                                       disp=disp, cov_type=cov_type,
-                                      cov_kwds=cov_kwds, use_t=use_t, **kwargs)
+                                      cov_kwds=cov_kwds, use_t=use_t,
+                                      max_start_irls=max_start_irls,
+                                      **kwargs)
 
     def _fit_gradient(self, start_params=None, method="newton",
                       maxiter=100, tol=1e-8, full_output=True,
                       disp=True, scale=None, cov_type='nonrobust',
-                      cov_kwds=None, use_t=None, **kwargs):
+                      cov_kwds=None, use_t=None, max_start_irls=3,
+                      **kwargs):
         """
         Fits a generalized linear model for a given family iteratively
         using the scipy gradient optimizers.
         """
 
+        if max_start_irls > 0:
```

Comment:
My thinking was the if there are user-provided start_params then they would
be starting values for the IRLS, and if the user wants it to go straight to
gradient they would set max_start_irls=0.

On Tue, Jan 27, 2015 at 12:41 PM, Josef Perktold <notifications@github.com>
wrote:

> In statsmodels/genmod/generalized_linear_model.py
> <https://github.com/statsmodels/statsmodels/pull/2204#discussion_r23626382>
> :
>
> >          """
> >          Fits a generalized linear model for a given family iteratively
> >          using the scipy gradient optimizers.
> >          """
> >
> > +        if max_start_irls > 0:
>
> this needs to check for user provided start_params which shouldn't be
> overwritten
> if start_params is None and max_start_irls > 0:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/statsmodels/statsmodels/pull/2204/files#r23626382>.
>